crates/polkavm-common/src/assembler.rs

```diff
@@ -35,6 +35,37 @@ fn parse_indirect_memory_access(text: &str) -> Option<(Reg, i32)> {
     }
 }
 
+/// Parses the long form of load_imm_and_jump_indirect:
+/// `tmp = {base}, {dst} = {value}, jump [tmp + {offset}]`, where `dest == base` is allowed
+fn parse_load_imm_and_jump_indirect(line: &str) -> Option<(Reg, Reg, i32, i32)> {
```

Comment:
```suggestion
fn parse_load_imm_and_jump_indirect_with_tmp(line: &str) -> Option<(Reg, Reg, i32, i32)> {
```

Since this can only handle the `tmp` form. Kinda long name, but I don't have any better ideas.

Reply:
yes, it also crossed my mind, but wasn't sure about about it. My other idea was `parse_load_imm_and_jump_indirect_long_form` but this is even longer.